xarray/backends/api.py

```diff
@@ -314,15 +316,15 @@ def _chunk_ds(
     chunked_array_type,
     from_array_kwargs,
     **extra_tokens,
-):
+) -> T_Dataset:
     chunkmanager = guess_chunkmanager(chunked_array_type)
 
     # TODO refactor to move this dask-specific logic inside the DaskManager class
     if isinstance(chunkmanager, DaskManager):
         from dask.base import tokenize
 
         mtime = _get_mtime(filename_or_obj)
-        token = tokenize(filename_or_obj, mtime, engine, chunks, **extra_tokens)
+        token: Any = tokenize(filename_or_obj, mtime, engine, chunks, **extra_tokens)
```

Comment:
```suggestion
        token: str = tokenize(filename_or_obj, mtime, engine, chunks, **extra_tokens)
```

This *should* be str

Reply:
Most of the times probably. But check a little below, token is redefined as `token = (None,)`.
Any was a lazy solution, since I remember tokenize being difficult (https://github.com/pydata/xarray/pull/8797)